storybookjs/storybook
# Svelte: Fix docs generating when using `lang="ts"` or optional chaining

Closes #24008 #24137






## What I did

When sveltedoc-parser fails to parse a Svelte component, Storybook was not injecting the documentation metadata.
This metadata was used to determine the component name. If the information is missing, then we use `component.name`, which is not always good. For example, when using vite and HMR, svelte generate a proxy named `Proxy<Component>` and Storybook was displaying this name instead of the real component name.

This PR always injects documentation metadata, with the real name of the component (build at compile time).

Moreover, sveltedoc-parser is configured to parse ecmascript v9 and fails to parse elvis operator, or "??". This PR also change sveltedoc-parser to enable ecmascript v12. However, the configuration is not officially exposed so I have to patch the lib.. I have proposed a real PR to the author of the library.

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_



1. Generate a svelte-vite sandbox
2. Start the storybook in dev mode
3. Modify "Button.svelte" and add "??" or "?." somewhere
4. The source generated for the stories should be valid. Without this PR, it shows `<Proxy<Button> .../>` instead.

### Documentation



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
